test/e2e/cypress/pageObject/base_po.js

```diff
@@ -71,13 +71,25 @@ export const clickUserDropdownProfileButton = () => {
   return cy.get(userDropdownProfileButton).click();
 };
 
-export const typeTotpCode = (totpSecret, inputField) => {
-  const { otp } = TOTP.generate(totpSecret);
-  return cy
-    .get(inputField)
-    .clear()
-    .type(otp)
-    .then(() => totpSecret);
+export const typeNextGeneratedTotpCode = (
+  totpSecret,
+  inputField,
+  forceNext = false
+) => {
+  const currentTime = Date.now();
```

Comment:
We could create an own function for this, something like getTotpWaitTime which return basicly the calculated time , just as an idea 
So in the end you have something like this
```
  const timeToWait = getTotpWaitTime(forceNext);
  
   return cy.wait(timeToWait).then(() => {
   ....
  ```